### PR TITLE
SwiftUI text view examples

### PR DESCRIPTION
**Pull request by suragch**
I've added some Text view examples to the SwiftUI gallery app. We can probably put my subcategories directly under the Text section of `PrimitivePage` but I didn't want to disturb the original example until I was sure this is what you wanted.

The attributed text section looks like it will be fairly large so I made a category for it but didn't start on it yet. 

I left localization and accessibility for Matteo and Vince. (I'm not sure what their GitHub names are or I would tag them.)

Let me know if I should be doing different when submitting pull requests.

Screenshot 2023-12-11 at 09 50 14
Screenshot 2023-12-11 at 09 50 32
Screenshot 2023-12-11 at 09 50 49
Screenshot 2023-12-11 at 09 51 05
Screenshot 2023-12-11 at 09 51 18
Screenshot 2023-12-11 at 09 51 31
Screenshot 2023-12-11 at 09 52 47

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@suragch thanks for posting this. Can you also provide screenshots in this PR of the examples you added? That helps contextualize the code changes.

**Comment by matthew-carroll** [comment]
@suragch can you let me know what lead to your particular breakdown of page titles? For example: initialization, options, styles?

Given that we have UX responsibilities broken up into particular categories, e.g., typography, localization, accessibility, etc, I'm wondering if it would be more effective to break down the example files that way? By using our categories, we can help prevent UX leads from stepping on each others' code as we ramp up the velocity.

**Comment by suragch** [comment]
@matthew-carroll
Screenshots added. Thanks for the reminder. I need to reread the contributing documentation. I remember you saying that now.

> can you let me know what lead to your particular breakdown of page titles? For example: initialization, options, styles?

I was trying to generally follow the SwiftUI API docs outline while not putting everything into one large scroll view. I agree with your idea of breaking them out into their respective UX responsibilities. I don't know if that will work for every kind of view but i think it will work for Text. Should I make three different categories under Text then: typography, localization, and accessibility? And do you think it would be better to further subdivide the typography examples into topics like styling and fitting or just to put everything on one screen with a scroll view?

**Comment by matthew-carroll** [comment]
> Should I make three different categories under Text then: typography, localization, and accessibility?

I think that's probably best.

> And do you think it would be better to further subdivide the typography examples into topics like styling and fitting or just to put everything on one screen with a scroll view?

I'll leave this up to you. Either way is fine. I'd say choose whichever one seems most likely to help with analysis and implementation.

**Comment by suragch** [comment]
@matthew-carroll 
I've reorganized the project structure with a folder for Typography, Localization, and Accessibility:

Screenshot 2023-12-11 at 15 03 04

Screenshot 2023-12-11 at 14 56 44

For now I am putting all of the typography examples into one long scroll view. We can further subdivide them later if necessary. This is the current content of the typography page 
Screenshot 2023-12-11 at 14 57 34
examples:

Screenshot 2023-12-11 at 14 57 08
Screenshot 2023-12-11 at 14 57 51
Screenshot 2023-12-11 at 14 58 12
Screenshot 2023-12-11 at 14 58 29

I moved a few of my previous localization related examples to the Localization page. @Maatteogekko feel free to delete or change them as needed:

Screenshot 2023-12-11 at 15 13 36

## Questions

- Since you added me to the repo as a collaborator, should I still make PRs or should I just push directly to the repo?
- Are the SwiftUI gallary examples meant to demonstrate UI or API or both?
- Attributed strings can duplicate a lot of the styling that is already available in the Text view. Are we trying to clone attributed strings also? My guess is no. Should the gallery app demonstrate everything that the [attributed string documentation](https://developer.apple.com/documentation/foundation/attributedstring) shows?

**Comment by suragch** [comment]
To answer one of my questions, I see we are planning to use [attributed_text](https://pub.dev/packages/attributed_text) package to replace attributed strings.

**Comment by matthew-carroll** [comment]
> Since you added me to the repo as a collaborator, should I still make PRs or should I just push directly to the repo?

Please use PRs. In general, no one should push directly to `main`.

> Are the SwiftUI gallary examples meant to demonstrate UI or API or both?

Both. We want to provide whatever devs are likely to need for implementation.

File: doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/localization/TextLocalizationPage.swift
Comment (matthew-carroll) [inline doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/localization/TextLocalizationPage.swift:5]: Let's not commit personal comments into the code base. Please use Discord, email, GitHub comments, etc. to communicate with other team members.

File: doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/localization/TextLocalizationPage.swift
Comment (matthew-carroll) [inline doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/localization/TextLocalizationPage.swift:1]: Is there any reason that we have empty comment lines at the beginning of the files?

File: doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/typography/TextTypographyPage.swift
Comment (matthew-carroll) [inline doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/typography/TextTypographyPage.swift:104]: I don't know if this TODO was from me previously, or from you, but let's include a bit more context as to why the work hasn't been done yet, what it's blocked on, etc.

File: doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/typography/TextTypographyPage.swift
Comment (matthew-carroll) [inline doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/typography/TextTypographyPage.swift:35]: Is there any reason for so many blank lines up here? One blank line is fine, but generally I don't expect to see 3 blank lines.

File: doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/typography/TextTypographyPage.swift
Comment (matthew-carroll) [inline doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/typography/TextTypographyPage.swift:208]: As a matter of consistency, let's capitalize the first character of any comment.

File: doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/typography/TextTypographyPage.swift
Comment (matthew-carroll) [inline doc_swift_ui_gallery/swift_ui_gallery/swift_ui_gallery/primitives/text/typography/TextTypographyPage.swift:252]: Is there a reason for this blank comment line?

**Comment by suragch** [comment]
@matthew-carroll Your review comments should be resolved now. Here are a couple of followup questions:

- I temporarily removed the attributed strings TODO, but should we add a whole range of attributed string examples?
- Some API items require iOS 17. Should we specify that at the line, file or project level (is that possible)? I assume the project level would be best if that's possible.

**Comment by matthew-carroll** [comment]
> I temporarily removed the attributed strings TODO, but should we add a whole range of attributed string examples?

We'll definitely want all relevant examples for attributed strings. The first question is which UX area that falls into. Which area do you think makes the most sense? Even if it's in typography, I'm totally fine with doing that in a followup PR. Let's just make sure an issue ticket exists for that.

> Some API items require iOS 17. Should we specify that at the line, file or project level (is that possible)? I assume the project level would be best if that's possible.

Do we need to note this at all? Is there something special about iOS 17? Every API came into existence at some given API level, but we haven't noted those earlier API numbers anywhere, so I'm curious why we'd want to note this one.

**Comment by suragch** [comment]
> We'll definitely want all relevant examples for attributed strings. The first question is which UX area that falls into. Which area do you think makes the most sense?

Probably typography. Let's do it in a future PR, though, so that we can unblock Mateo and Vince from starting their work.

> Do we need to note this at all? Is there something special about iOS 17?

Currently Xcode gives an error for those items I marked as iOS 17 (uncomment the code to see). I can fix the error locally by adding an annotation, but I'm wondering if there is a project level way of saying that we want to accept all the newest APIs.

**Comment by matthew-carroll** [comment]
> I can fix the error locally by adding an annotation, but I'm wondering if there is a project level way of saying that we want to accept all the newest APIs.

Unfortunately, I don't know the answer to this. I don't have a lot of direct XCode iOS experience. I'd say let's do whatever allows people that aren't on the bleeding edge to still compile and run the project.

**Comment by matthew-carroll** [comment]
@suragch let me know if you think the PR is ready for a final review. I wasn't sure if you wanted to make changes based on the last comment, or leave it as is.

**Comment by suragch** [comment]
> I'd say let's do whatever allows people that aren't on the bleeding edge to still compile and run the project.

I found a way to optionally show the iOS 17 related views using `if #available(iOS 17.0, *)`. This will allow people without access to iOS 17 to still run the app. The following three Text views won't show for them though:

Screenshot 2023-12-14 at 09 23 00

I think the PR is ready to merge now.

**Review by matthew-carroll — APPROVED** [review]
LGTM

**Pull request merged**
